distributed/worker.py

```diff
@@ -535,7 +541,11 @@ def __init__(
                 "distributed.worker.memory.pause"
             )
 
-        if isinstance(data, MutableMapping):
+        if isinstance(data, MutableMapping) and not isinstance(data, TypeCompressor):
+            self.data = TypeCompressor()
+            for k, v in data.values():
+                self.data[k] = data[k]
+        elif isinstance(data, TypeCompressor):
```

Comment:
I'm not sure what's going on here.  Was it necessary to special-case `TypeCompressor` here?  Ideally we only depend on the MutableMapping interface and don't have to worry too much about every possible type that someone might provide individually.  That can make future maintenance an issue.